src/librustc_codegen_ssa/base.rs

```diff
@@ -117,30 +117,61 @@ pub fn compare_simd_types<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
 ///
 /// The `old_info` argument is a bit odd. It is intended for use in an upcast,
 /// where the new vtable for an object will be derived from the old one.
-pub fn unsized_info<'tcx, Cx: CodegenMethods<'tcx>>(
-    cx: &Cx,
+pub fn unsized_info<'tcx, 'a, Bx: BuilderMethods<'a, 'tcx>>(
```

Comment:
Some ancient context for this being where it is, and only taking `Cx`: it was shared between the old codegen logic (based off of AST/HIR, pre-MIR), MIR-based codegen, but also some constant stuff that is now all in miri.

We should've moved this into a better place a long time ago.
Before this PR, it should probably be in miri, since it can only ever return a constant, *except* for the case when `old_info` is used - which is the case in which this PR changes anything anyway.